Azure/azure-cli
# [Core] WAM integration

**Related command**

* `az login`
* And any subsequent command that would obtain a token via MSAL's `acquire_token_silent()`.

**Description**

Close #21201
Require https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/415

This PR is currently a draft to demonstrate how Azure CLI could enable the broker behavior from an MSAL feature branch. 

**Testing Guide**

1. Run the following commands on a Windows 10+ machine
2. Somehow install Azure CLI from this feature branch, conceptually similar to this (however this command does not currently work)
   `pip install git+https://github.com/azure/azure-cli@broker-integration`
3. Then run `az login`, and notice how a pop-up window (rather than a browser) would pop up.
4. Try run some more (all?) different `az foo bar` command to make sure the token returned by broker would work


**History Notes**



---

This checklist is used to make sure that common guidelines for a pull request are followed.
